tests/tests_pytorch/trainer/optimization/test_optimizers.py

```diff
@@ -657,3 +658,67 @@ def lr_scheduler_step(*_): ...
     else:
         with pytest.raises(MisconfigurationException, match="CustomScheduler` doesn't follow"):
             _init_optimizers_and_lr_schedulers(model)
+
+
+@patch("torch.optim.lr_scheduler.StepLR.step")
+def test_lr_scheduler_step_across_epoch_boundaries(mocked_sched, tmp_path):
+    class StepAcrossEpochsModel(LightningModule):
+        def __init__(self):
+            super().__init__()
+            self.layer = torch.nn.Linear(32, 2)
+
+        def forward(self, x):
+            return self.layer(x)
+
+        def training_step(self, batch, batch_idx):
+            # Add print statement to track batch index and global step
+            if hasattr(self, 'trainer'):
+                print(f"Batch idx: {batch_idx}, Global step: {self.trainer.global_step}")
```

Comment:
Print statements in tests are not super helpful, just use `assert`s so the test will break if we don't get the expected value here.